core/src/main/java/com/graphhopper/routing/util/parsers/OSMMaxHeightParser.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util.parsers;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.profiles.EncodedValue;
+import com.graphhopper.routing.profiles.EncodedValueLookup;
+import com.graphhopper.routing.profiles.MaxHeight;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.storage.IntsRef;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class OSMMaxHeightParser implements TagParser {
+
+    private final DecimalEncodedValue heightEncoder;
+    private final boolean enableLog;
+
+    public OSMMaxHeightParser() {
+        this(MaxHeight.create(), false);
+    }
+
+    public OSMMaxHeightParser(DecimalEncodedValue heightEncoder, boolean enableLog) {
+        this.heightEncoder = heightEncoder;
+        this.enableLog = enableLog;
+    }
+
+    @Override
+    public void createEncodedValues(EncodedValueLookup lookup, List<EncodedValue> registerNewEncodedValue) {
+        registerNewEncodedValue.add(heightEncoder);
+    }
+
+    @Override
+    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, EncodingManager.Access access, long relationFlags) {
+        List<String> heightTags = Arrays.asList("maxheight", "maxheight:physical"/*, do not use "height" for way height! */);
```

Comment:
> do not use "height" for way height!

What do you mean by that?

Reply:
the height in OSM != height of a street. Unlike width.

Reply:
Ah, thanks for the explanation. I wasn't sure if this would be related to elevation. Could we maybe change the comment slightly, like:

` the OSM tag "height" is not used for maxheight` 

or something like this?